crates/services/gas_price_service/simulation/src/data_sources.rs

```diff
@@ -0,0 +1,116 @@
+use async_trait::async_trait;
+use fuel_core_gas_price_service::{
+    common::{
+        l2_block_source::L2BlockSource,
+        utils::BlockInfo,
+    },
+    v1::{
+        algorithm::SharedV1Algorithm,
+        da_source_service::{
+            service::DaBlockCostsSource,
+            DaBlockCosts,
+        },
+    },
+};
+
+use fuel_core_gas_price_service::common::utils::{
+    Error as GasPriceError,
+    Result as GasPriceResult,
+};
+use fuel_core_types::fuel_types::BlockHeight;
+
+pub struct SimulatedL2Blocks {
+    recv: tokio::sync::mpsc::Receiver<BlockInfo>,
+    shared_algo: SharedV1Algorithm,
+}
+
+impl SimulatedL2Blocks {
+    pub fn new(
+        recv: tokio::sync::mpsc::Receiver<BlockInfo>,
+        shared_algo: SharedV1Algorithm,
+    ) -> Self {
+        Self { recv, shared_algo }
+    }
+
+    pub fn new_with_sender(
+        shared_algo: SharedV1Algorithm,
+    ) -> (Self, tokio::sync::mpsc::Sender<BlockInfo>) {
+        let (send, recv) = tokio::sync::mpsc::channel(16);
+        (Self { recv, shared_algo }, send)
+    }
+}
+
+#[async_trait]
+impl L2BlockSource for SimulatedL2Blocks {
```

Comment:
Unrelated to this PR; Do we still need `async_trait`? These are our traits right? We shouldn't need the legacy `async_trait` macro anymore since async functions in traits have been stabilized.

Reply:
you're right